cdap-ui/app/cdap/components/DataPrep/Directives/MapToTarget/index.tsx

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useEffect, useState } from 'react';
+import T from 'i18n-react';
+import { makeStyles } from '@material-ui/core';
+import classnames from 'classnames';
+import { preventPropagation, connectWithStore } from 'services/helpers';
+import { setPopoverOffset } from 'components/DataPrep/helper';
+import { CurrentSelection } from 'components/DataPrep/Directives/MapToTarget/CurrentSelection';
+import { LoadingBar } from 'components/DataPrep/Directives/MapToTarget/LoadingBar';
+import { OptionFilter } from 'components/DataPrep/Directives/MapToTarget/OptionFilter';
+import { OptionList } from 'components/DataPrep/Directives/MapToTarget/OptionList';
+import IconSVG from 'components/IconSVG';
+import DataPrepStore, { IDataModel, IModel, IModelField } from 'components/DataPrep/store';
+import {
+  execute,
+  setError,
+  loadTargetDataModelStates,
+  saveTargetDataModelStates,
+  setTargetDataModel,
+  setTargetModel,
+} from 'components/DataPrep/store/DataPrepActionCreator';
+
+const PREFIX = 'features.DataPrep.Directives.MapToTarget';
+
+const useStyles = makeStyles({
+  clearfix: {
+    '&::after': {
+      display: 'block',
+      clear: 'both',
+      content: 'none',
+    },
+  },
+  floatRight: {
+    float: 'right',
+  },
+});
+
+interface IMapToTargetProps {
+  isOpen: boolean;
+  isDisabled: boolean;
+  column: string;
+  onComplete: () => void;
+  close: () => void;
+  dataModelList?: IDataModel[];
+  targetDataModel?: IDataModel;
+  targetModel?: IModel;
+}
+
+const MapToTarget = (props: IMapToTargetProps) => {
+  const classes = useStyles(undefined);
+
+  const {
+    isOpen,
+    isDisabled,
+    column,
+    onComplete,
+    close,
+    dataModelList,
+    targetDataModel,
+    targetModel,
+  } = props;
+
+  const [loadingText, setLoadingText] = useState('');
+  const [searchText, setSearchText] = useState('');
+
+  useEffect(() => {
+    let pending = true;
+    setLoadingText(`${PREFIX}.initializingText`);
+
+    const initialize = async () => {
+      try {
+        await loadTargetDataModelStates();
+      } catch (error) {
+        setError(error);
```

Comment:
This seems to be not useful. This is what I am seeing if the wrangler app is not configured properly to serve this request.
<img width="1440" alt="Screen Shot 2020-05-12 at 3 31 21 PM" src="https://user-images.githubusercontent.com/1452845/81752814-25816200-9467-11ea-9165-2854dee25d0b.png">

Reply:
Can you please provide details how to reproduce that?

Reply:
I was able to reproduce it using wrangler app that does not have a valid handler for adding data models.

Reply:
It looks like the issue was fixed by adding check for `error.response` to `setError()` function. Anyway, I added prefix message saying **Map to Target directive initialization failed**.